docs/blueprints/linux-setup.md

```diff
@@ -0,0 +1,181 @@
+
+All these distributions are shipped with systemD as a service manager.
+
+
+
+
+## Design
+The installer will be a debian package. We choose to auto-update the binary, even if this will make the package not following debian guidelines
+- When installing the package, it should prompt for user input for:
+  - Wallet address
+  - Email
+  - External address/port
+  - Advertised bandwidth
+  - Advertised storage
+  - Identity directory
+  - Installation directory
+  - Storage directory
+- Generate `config.yaml` file with the user configuration.
+
+We choose to reuse the storagenode-updater and the updater-updater used in windows. They will be daemonized using systemD. The storagenode updater will auto-update. A recovery will be triggered if the updated updater service fails to restart.
+We will use debconf to retrieve user data. 
+
+The debian package will NOT contain the storagenode and storagenode-updater binaries. They will be downloaded as part of the post-installation script. A separate git repository will be created for holding the debian package.
+
+Once we get a fully working debian package, we can convert it to the RPM format using the fpm tool. There are no debconf-like for RPMs, we will need to implement a post-install script to gather user inputs.
+
+The debian package will be available by direct download and on a APT repository that users can add to their package manager source list. The repository will be managed using reprepro. Each time the repository is modified, it commits the static content to a dedicated git repository. 
+
+## Rationale
+
+### storagenode service
+As stated earlier, systemD is the commonly used service manager. It is the default on raspbian, debian, ubuntu, redhat, archlinux.
+Hence, we should use systemD for building our storagenode service. 
+
+### Installation
+#### Custom installer
+Packaging in its simplest form would be tar.gz with an installation binary. This solution would be simple for us, but represents an annoyance for the user as our application would not be managed by their package manager.
+
+#### Packages
+A package is an archive file containing the application and metadata for indicating to the package manager how to install it. 
+Its format depends on the used package manager.
+The most common formats are:
+- deb for debian-based distributions
+- rpm for red hat-based distributions
+- .tar.xz for arch
+
+Guidelines about how to build packages are provided for the most commonly used package formats:
+- deb: https://go-team.pages.debian.net/packaging.html
+- rpm: https://docs.fedoraproject.org/en-US/packaging-guidelines/Golang/
+- arch: https://wiki.archlinux.org/index.php/Go_package_guidelines
+
+
+The process for building a package is as follows:
+    - make a source package
+    - compile it to get binary packages.
+Only the binary package is used by the user for installation. It is not a recommended pratice to directly integrate binaries.
```

Comment:
This part does not render well in markdown. I believe we need an empty line before and after the bullet list.